### ISSUE TITLE
Wrong account making "This change has been released" notes on issues

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1933]
@afakebot used to make these announcements, but now it's the github actions bot

as seen in https://github.com/FakeItEasy/FakeItEasy/issues/1925#issuecomment-1520618129

(see https://github.com/FakeItEasy/FakeItEasy/issues/1848#issuecomment-910250775 for the preferred version). We must've used the wrong token when converting to GitHub actions from AppVeyor

**Comment by thomaslevesque** [comment]
The code reads the `GITHUB_TOKEN` environment variable to get the token. But the workflow sets the `GH_TOKEN` variable instead... So we end up using the default token provided by Github Actions.

**Comment by thomaslevesque** [comment]
OK, I completely misunderstood what was happening. I was looking at the wrong step.
The step where we set `GH_TOKEN` is the one that triggers the docs generation using the GitHub CLI (`gh workflow run...`).
In the step that calls the deployment script, we do set the `GITHUB_TOKEN` environment variable, but we set it to `${{ github.token }}`, which is the "GitHub actions bot" token. We just need to set it to the `BOT_GITHUB_TOKEN` secret instead.
